### PR TITLE
Resolved issue with View Type preference not properly encoded

### PR DESCRIPTION
**Pull request by matthewjohns0n**

### CODE REVIEW

File: system/ee/ExpressionEngine/Service/File/ViewType.php [hunk 1]
```diff
@@ -27,9 +27,8 @@ public function determineViewType($destination = 'all', $viewtype = 'list')
     {
         $views = ['list', 'thumb'];
         $viewtype_prefs = [];
-
         if (ee()->input->cookie('viewtype')) {
-            $viewtype_prefs = unserialize(ee()->input->cookie('viewtype'));
+            $viewtype_prefs = json_decode(ee()->input->cookie('viewtype'), true);
```
Comment (bryannielsen) [inline system/ee/ExpressionEngine/Service/File/ViewType.php:31]: If we were previously serializing these cookies do we need some error handling around this?  I'm not sure how long the cookies live for but I imagine it's possible we could be trying to json_decode a serialized value during the transition?

Comment (intoeetive) [reply system/ee/ExpressionEngine/Service/File/ViewType.php:31]: I agree with what Bryan is saying. We had similar issue with Entry Manager views (switching from serializing to JSON encoding or vice versa) and it was bad experience for people who had that saved already. We need some fallback, at least, if we can't avoid the change

Comment (matthewjohns0n) [reply system/ee/ExpressionEngine/Service/File/ViewType.php:31]: I was able to regex the settings out of the serialized string, and then json encode & set them in the cookie, so there is no loss of data at all.

**Review by intoeetive — APPROVED** [review]
worked well for me after fixing the typo

**Review by bryannielsen — APPROVED** [review]

**Pull request merged**